doc/decisions/0010-dynamic-answer-options.md

```diff
@@ -0,0 +1,256 @@
+
+### Dynamic answer options with repeating answer source within a repeat (Checkbox)
+
+```json
+{
+  "id": "answer-id",
+  "mandatory": true,
+  "type": "Checkbox",
+  "dynamic_options": {
+    "values": {
+      "source": "answers",
+      "identifier": "checkbox-answer-id",
+      "list_item_selector": {
+        "source": "location",
+        "id": "list_item_id"
+      }
+    },
+    "transform": {
+      "option_label_from_value": [
+        "self",
+        "checkbox-answer-id",
+        {
+          "source": "location",
+          "identifier": "list_item_id"
+        }
+      ]
+    }
+  },
+  "options": [
+    {
+      "label": "None of the above",
+      "value": "None of the above"
+    }
+  ]
+}
+```
+
+- `values.list_item_selector` is not strictly needed. Dynamic options code could always pass the current list item id, and there is already logic in our value source resolver to use that list item id if the answer id is in a repeating section or a list collector block.
+
+### Dynamic answer options with repeating answer source outside a repeat (Checkbox)
+
+```json
+{
+  "id": "answer-id",
+  "mandatory": true,
+  "type": "Checkbox",
+  "dynamic_options": {
+    "values": {
+      "source": "answers",
+      "identifier": "checkbox-answer-id",
+      "list_item_selector": {
+        "source": "list",
+        "id": "people",
+        "id_selector": "first"
+      }
+    },
+    "transform": {
+      "option_label_from_value": [
+        "self",
+        "checkbox-answer-id",
+        {
+          "source": "list",
+          "id": "people",
+          "id_selector": "first"
+        }
+      ]
+    }
+  },
+  "options": [
+    {
+      "label": "None of the above",
+      "value": "None of the above"
+    }
+  ]
+}
+```
+
+### Dynamic answer options with list source
+
+```json
+{
+  "id": "answer-id",
+  "mandatory": true,
+  "type": "Checkbox",
+  "dynamic_options": {
+    "values": {
+      "source": "list",
+      "identifier": "people",
+      "id_selector": "items"
+    },
+    "transform": {
+      "concat": [
+        [
+          {
+            "source": "answers",
+            "identifier": "first-name"
+          },
+          {
+            "source": "answers",
+            "identifier": "last-name"
+          }
+        ],
+        " "
+      ]
+    }
+  },
+  "options": [
+    {
+      "label": "None of the above",
+      "value": "None of the above"
+    }
+  ]
+}
+```
+
+- The `concat` operator is responsible for concatenating a sequence of values using a delimiter.
+  This example demonstrates formatting the answer values for each list item.
+- When dynamic options are used with a `list` source, the `list_item_id` is not passed explicitly as `self` as the call is responsible for using the rule evaluator in the context of the current list item id.
```

Comment:
This isn't completely clear to me..

> as the call is responsible for using the rule evaluator in the context of the current list item id

..does this mean the `concat` transform itself has the logic to retrieve the `first-name` and `last-name` from the answer store?